app/components/point-input.jsx

```diff
@@ -0,0 +1,191 @@
+'use strict'
+import React, { useEffect, useRef, useState } from 'react';
+import { createUseStyles } from 'react-jss';
+import autosize from 'autosize';
+
+function PointInput(props) {
+    const { style={}, className="", maxWordCount = 30, maxCharCount = 100, defaultValue = {description: "", subject: ""}, onDone = (valid=false, values = { subject: "", description: "" }) => ({valid, ...values}) } = props
+    const classes = useStyles()
+    const [subject, setSubject] = useState(defaultValue?.subject ?? "")
+    const [description, setDescription] = useState(defaultValue?.description ?? "")
+    const [descWordCount, setDescWordCount] = useState(getDescWordCount(description))
+    const [subjCharCount, setSubjCharCount] = useState(getSubjCharCount(subject))
+    const [isSubjFocused, setIsSubjFocused] = useState(false)
+    const [isDescFocused, setIsDescFocused] = useState(false)
+    const textareaRef = useRef(null);
+
+    useEffect(() => {
+        autosize(textareaRef.current)
+
+        return () => {
+            autosize.destroy(textareaRef.current)
+        }
+    }, [])
+
+    function getDescWordCount(inputText) {
+        if (!inputText) return 0
+        return inputText.trim().split(/\s+/).length;
+    }
+
+    function getSubjCharCount(inputText) {
+        return inputText.length
+    }
+
+    const isDescValid = inputText => {
+        const wordCount = getDescWordCount(inputText)
+        return wordCount > 0 && wordCount <= maxWordCount
+    }
+
+    const isSubjValid = inputText => {
+        const charCount = getSubjCharCount(inputText)
+        return charCount > 0 && charCount <= maxCharCount
+    }
+
+    const handleSubjectChange = (value) => {
+        setSubject(value)
+        setSubjCharCount(getSubjCharCount(value))
+    }
+
+    const handleDescriptionChange = (value) => {
+        setDescription(value)
+        setDescWordCount(getDescWordCount(value))
+    }
+
+    const handleDescriptionBlur = () => {
+        setIsDescFocused(false)
+        onDone((isSubjValid(subject) && isDescValid(description)), ({ subject: subject, description: description }))
+    }
+
+    return (
+        <>
+            <div className={classes.container}>
```

Comment:
Should follow the style and be something like:
```
<div className={cx(classes.container), className)} {...otherProps} />
```
see https://github.com/EnCiv/civil-pursuit#react-component-guidelines-and-notes